app/scripts/views/mixins/signed-out-notification-mixin.js

```diff
@@ -7,38 +7,30 @@
 define(function (require, exports, module) {
   'use strict';
 
-  const BaseView = require('views/base');
   const Notifier = require('lib/channels/notifier');
   const Session = require('lib/session');
+  const Url = require('lib/url');
 
   var Mixin = {
     notifications: {
       // populated below using event name aliases
     },
 
     clearSessionAndNavigateToSignIn () {
-      // Unset uid otherwise it will henceforth be impossible
-      // to sign in to different accounts inside this tab.
-      this.relier.unset('uid');
-      this.relier.unset('email');
       this.user.clearSignedInAccountUid();
-      if (this._formPrefill) {
-        // The user has manually signed out, a pretty strong indicator
-        // the user does not want any of their information pre-filled
-        // on the signin page. Clear any remaining formPrefill info
-        // to ensure their data isn't sticking around in memory.
-        this._formPrefill.clear();
-      }
       Session.clear();
       this.navigateToSignIn();
     },
 
     navigateToSignIn () {
-      this.navigate('signin', {
-        success: BaseView.t('Signed out successfully')
-      }, {
-        clearQueryParams: true
+      const queryString = Url.objToSearchString({
+        context: this.relier.get('context'),
```

Comment:
Not including `migration` I understand, though I think we should continue to propagate the `entrypoint` and `utm_` query params. One user was signed out, and the next signed in, the entrypoint is still the same, and we'll want to know where the new users came from when they complete the sign in.

As an aside, can you add a note about why only certain query parameters and not the query parameter list verbatim?